### DISCUSSION TITLE
Consider separating `tokenize`/`normalize_token` into its own library

### DISCUSSION THREAD

**Discussion by timkpaine** [dask/dask#11370] [Ideas]
`dask`'s `tokenize` function is extremely useful and has a wide array of applications outside of dask: https://github.com/dask/dask/blob/4be4880bce4cca6374c5dc90534c1117e20e406f/dask/base.py#L1019

It is also relatively standalone, and the default `normalize_token` registers many builtins: https://github.com/dask/dask/blob/4be4880bce4cca6374c5dc90534c1117e20e406f/dask/base.py#L1112

Would it make sense to separate into its own repo, given it has no real dependencies and could be leveraged (along with the `__dask_tokenize__` interface) in many more libraries that might not want a full dask dependency?

**Comment by mrocklin** [comment]
> that might not want a full dask dependency

Core Dask is intentionally very lightweight for this reason.  I suspect that if you looked at the core dependencies of the project you'd find that there was nothing there that worried you.

Note that `pip install dask` is quite different than `pip install dask[complete]` and `conda install dask-core` is quite different than `conda install dask`
